test/session_tests/DocumentQueryTest.ts

```diff
@@ -8,220 +8,289 @@ import {RequestExecutor} from "../../src/Http/Request/RequestExecutor";
-    });
+  it('should query by startsWith', async () => {
+    const results: Universal[] = await store.openSession()
+      .query<Universal>({
+        indexName: 'Universals',
+        WaitForNonStaleResults: true
+      })
+      .startsWith<string>('name','wi').get();
+    expect(results[0]).to.be.a('object');
+  });
 
-    it('should paginate', async() => {
-      const expectedUids: number[][] = [[2,3],[4,5],[6,90],[95]];
-      const pageSize: number = 2;
-      
-      const totalCount: number = await store.openSession().query<Product>({
-        documentType: Product,
-        waitForNonStaleResults: true
-      }).whereNotNull('uid').count();
-
-      const totalPages: number = Math.ceil(totalCount / pageSize);
-
-      expect(totalCount).to.equals(7);
-      expect(totalPages).to.equals(4);
-
-      for (let page: number = 1; page <= totalPages; page++) {
-        const products: Product[] = await store.openSession().query<Product>({
-          documentType: Product,
-          waitForNonStaleResults: true
-        }).whereNotNull('uid').orderBy('uid').skip((page - 1) * pageSize).take(pageSize).get();
-
-        expect(products).to.have.length.lte(pageSize);
-        products.forEach((product: Product, index: number) => expect(product.uid).to.equals(expectedUids[page - 1][index]));
-      }
-    });
+  it('should query by endsWith', async () => {
+    const results: Universal[] = await store.openSession()
+      .query<Universal>({
+        indexName: 'Universals',
+        WaitForNonStaleResults: true
+      })
+      .endsWith<string>('name','ing').get();
+    expect(results[0]).to.be.a('object');
+  });
 
-    it('should query with includes', async() => {
-      session = store.openSession();
+  it('should query by search', async () => {
+    const results: Universal[] = await store.openSession()
+      .query<Universal>({
+        indexName: 'Universals',
+        WaitForNonStaleResults: true
+      })
+      .search('name', 'withNesting', 'order', 1, 2).get();
+    expect(results[0]).to.include({'name': 'withNesting'});
+  });
 
-      await session.query({
-        waitForNonStaleResults: true, 
-        includes: ['product_id']
-      }).where({uid: 92}).get();
+  it('should query with orderBy', async () => {
+    const results: Universal[] = await store.openSession()
+      .query<Universal>({
+        indexName: 'Universals',
+        WaitForNonStaleResults: true
+      })
+      .orderBy<string>('name', 'ASC').get();
+    expect(results[0]).to.be.a('object');
+  });
 
-      await session.load('Product/108');
-      expect(session.numberOfRequestsInSession).to.equals(1);            
-    });
+  it('should query with orderByDESC', async () => {
+    const results: Universal[] = await store.openSession()
+      .query<Universal>({
+        indexName: 'Universals',
+        WaitForNonStaleResults: true
+      })
+      .orderBy<string>('name', 'DESC').get();
+    expect(results[0]).to.be.a('object');
```

Comment:
It's orderBy test and you don't even check if the results you got are ordered in any way.